src/EhsnPlugin/Mappers/MeasurementParser.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using EhsnPlugin.DataModel;
+using EhsnPlugin.SystemCode;
+
+namespace EhsnPlugin.Mappers
+{
+    public class MeasurementParser
+    {
+        private readonly EHSN _eHsn;
+        private DateTime _visitDate;
+
+        public MeasurementParser(EHSN eHsn, DateTime visitDate)
+        {
+            _eHsn = eHsn ?? throw new ArgumentNullException(nameof(eHsn));
+
+            _visitDate = visitDate;
+        }
+
+        public EhsnMeasurement Parse()
+        {
+            var measurementMeanTime = GetMeasurementMeanTimeOrMinValue();
+
+            return new EhsnMeasurement
+            {
+                StageMeasurements = ParseStageMeasurements(),
+                DischargeMeasurements = ParseDischargeMeasurements(measurementMeanTime),
+                EnvironmentConditionMeasurements = ParseEnvironmentConditionMeasurements(measurementMeanTime),
+                SensorStageMeasurements = ParseMeasResults()
+            };
+        }
+
+        private DateTime GetMeasurementMeanTimeOrMinValue()
+        {
+            if (_eHsn.DisMeas == null)
+                return DateTime.MinValue;
+
+            //Start/End times should both exist in eHSN files:
+            var start = ParseTimeOrMinValue(_eHsn.DisMeas.startTime);
+            var end = ParseTimeOrMinValue(_eHsn.DisMeas.endTime);
+
+            if (start == DateTime.MinValue || end == DateTime.MinValue)
+                return DateTime.MinValue;
+
+            return new DateTime((start.Ticks + end.Ticks) / 2);
```

Comment:
I was looking at a similar problem in the Stage/Discharge plugin today. Calculating the midpoint of an interval.

I ended up add some tweaking to truncate the midpoint to a pure minute boundary, and avoid weird fractional sections caused by the divide by 2. This will make the times more "human readable" when shown in the Field Data Editor.

Something like this:
```C#
var midpoint = new DateTime((start.Ticks + end.Ticks) / 2);

var humanTime = new DateTime(
    midpoint.Year,
    midpoint.Month,
    midpoint.Day,
    midpoint.Hour,
    midpoint.Minute,
    0);

return humanTime < start ? start : humanTime;
```

Reply:
Good point. Fixed as suggested.